api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerAuthenticationSslPlaintextVault.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public class KafkaListenerAuthenticationSslPlaintextVault extends KafkaListenerAuthentication {
+
+    public static final String SASL_PLAINTEXT = "sasl_plaintext";
+    private String vaultAddr;
+    private String vaultToken;
+    private String adminPath = "secret/kafka/private/admin";
+    private String usersPath = "secret/kafka/private/users";
+    private boolean cacheEnabled;
+    private int cacheTTL;
+
+    @Description("Must be `" + SASL_PLAINTEXT + "`")
+    @Override
+    public String getType() {
+        return SASL_PLAINTEXT;
+    }
+
+    @Description("Vault Address Server")
```

Comment:
I assume that some of these options will be mandatory. Could you add following annotation to those?

```
@JsonProperty(required = true)
```

To those which are optional, should have:

```
@JsonInclude(value = JsonInclude.Include.NON_NULL)
```

You can also add the default value

```
@JsonProperty(defaultValue = "false")
```